pinot-controller/src/main/resources/app/App.tsx

```diff
@@ -19,49 +19,87 @@
 
 import * as React from 'react';
 import * as ReactDOM from 'react-dom';
-import { MuiThemeProvider } from '@material-ui/core';
-import { Switch, Route, HashRouter as Router } from 'react-router-dom';
+import { CircularProgress, createStyles, makeStyles, MuiThemeProvider } from '@material-ui/core';
+import { Switch, Route, HashRouter as Router, Redirect } from 'react-router-dom';
 import theme from './theme';
 import Layout from './components/Layout';
 import RouterData from './router';
 import PinotMethodUtils from './utils/PinotMethodUtils';
 import CustomNotification from './components/CustomNotification';
 import { NotificationContextProvider } from './components/Notification/NotificationContextProvider';
+import app_state from './app_state';
+
+const useStyles = makeStyles(() =>
+  createStyles({
+    loader: {
+      position: 'fixed',
+      left: '50%',
+      top: '30%'
+    },
+  })
+);
 
 const App = () => {
   const [clusterName, setClusterName] = React.useState('');
+  const [loading, setLoading] = React.useState(true);
+
   const fetchClusterName = async () => {
     const clusterNameResponse = await PinotMethodUtils.getClusterName();
     localStorage.setItem('pinot_ui:clusterName', clusterNameResponse);
     setClusterName(clusterNameResponse);
   };
+
+  const fetchClusterConfig = async () => {
+    const clusterConfig = await PinotMethodUtils.getClusterConfigJSON();
+    app_state.queryConsoleOnlyView = clusterConfig?.queryConsoleOnlyView === 'true';
```

Comment:
Apologies for my lack of syntax understanding, but will this set the `app_state.queryConsoleOnlyView` flag to false if queryConsoleOnlyView is missing or is set to false in the config ?

Reply:
yes, that right. only if there is queryConsoleOnlyView and set as  "true" in string then it will be set to true else false.

Reply:
`app_state.queryConsoleOnlyView` will remain `undefined`/`false` if either of `clusterConfig` or `clusterConfig.queryConsoleOnlyView` is undefined, or if `queryConsoleOnlyView` is explicitly set to "false".

Reply:
ok sounds good. That was the expectation. Thanks for confirming.